internal/controller/controllers/agent_controller.go

```diff
@@ -588,11 +594,31 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 	} else {
 		setConditionsUnknown(agent)
 	}
+	// get node
+	var node *corev1.Node
+	if funk.Contains([]models.HostStage{models.HostStageRebooting, models.HostStageJoined, models.HostStageConfiguring}, agent.Status.Progress.CurrentStage) {
+		node, err = r.getNode(ctx, agent)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			r.Log.WithError(err).Errorf("agent %s/%s: failed to get node", agent.Namespace, agent.Name)
+			return ctrl.Result{}, err
+		}
+		if node != nil {
+			if isNodeReady(node) {
+				err = r.updateHostInstallProgress(ctx, h, models.HostStageDone)
+				agent.Status.Progress.CurrentStage = models.HostStageDone
+			} else {
+				err = r.updateHostInstallProgress(ctx, h, models.HostStageJoined)
+				agent.Status.Progress.CurrentStage = models.HostStageJoined
+			}
+			if err != nil {
+				r.Log.WithError(err).Errorf("Failed updating host %s install progress", h.ID)
+				return ctrl.Result{}, err
+			}
+		}
+	}
 	if shouldAutoApproveCSRs {
-		alreadyApproved := r.tryApproveDay2CSRs(ctx, agent)
-		if alreadyApproved {
-			agent.Status.Progress.CurrentStage = models.HostStageDone
-		} else {
+		alreadyApproved := r.tryApproveDay2CSRs(ctx, agent, node)
```

Comment:
Since only none platform day2 CSRs are auto approved, the logic for updating status/stage will not work correctly for (day2 , BM) since no requeue will be done for BM

Reply:
Why would we need to requeue?
The reconcile should get called the same way it get's called for approving the day2 CSRs

Reply:
Hmmm, I think you are right (though it does seems to work without the requeue), we should requeue the all agents once they reach status `Rebooting` until they reach `Done`  to make sure we keep reconciling them